packages/fx-core/src/common/azure-hosting/hostingFactory.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzureHosting } from "./azureHosting";
+import { FunctionHosting } from "./functionHosting";
+import { BotHosting } from "./botServiceHosting";
+import { HostType } from "./interface";
+
+const HostingMap: { [key: string]: () => AzureHosting } = {
+  [HostType.Function]: () => new FunctionHosting(),
+  [HostType.BotService]: () => new BotHosting(),
+};
+
+export class HostingResourceFactory {
```

Comment:
I would prefer the class name to be "AzureHostingFactory", because it is straightforward as the factory of AzureHosting.

Reply:
Updated.